server/chat-plugins/room-faqs.ts

```diff
@@ -111,16 +111,25 @@ export const commands: Chat.ChatCommands = {
 		if (!(roomFaqs[room.roomid] && roomFaqs[room.roomid][topic])) return this.errorReply("Invalid topic.");
 		if (
 			room.settings.repeats?.length &&
-			room.settings.repeats.filter(x => x.faq && x.id === (getAlias(room!.roomid, topic) || topic)).length
+			room.settings.repeats.filter(x => x.faq && x.id === topic).length
 		) {
-			this.parse(`/msgroom ${room.roomid},/removerepeat ${getAlias(room.roomid, topic) || topic}`);
+			this.parse(`/msgroom ${room.roomid},/removerepeat ${topic}`);
 		}
-		delete roomFaqs[room.roomid][topic];
 		Object.keys(roomFaqs[room.roomid]).filter(
 			val => getAlias(room!.roomid, val) === topic
 		).map(
-			val => delete roomFaqs[room!.roomid][val]
+			val => {
+				delete roomFaqs[room!.roomid][val];
+				if (
+					room?.settings.repeats?.length &&
+					room.settings.repeats.filter(x => x.faq && x.id === val).length
+				) {
+					this.parse(`/msgroom ${room.roomid},/removerepeat ${val}`);
+				}
+				return true;
+			}
 		);
```

Comment:
This should now be using a for-of loop instead of map, that's what the linter error was meant to encourage in the first place.